kythe/go/serving/pipeline/beam.go

```diff
@@ -76,33 +248,45 @@ func (k *KytheBeam) References() beam.PCollection {
 				facts.SnippetStart, facts.SnippetEnd,
 			},
 		}, k.nodes))
-	files := beam.ParDo(s, keyByPath, beam.ParDo(s,
-		&nodes.Filter{
-			FilterByKind: []string{kinds.File},
-			IncludeFacts: []string{facts.Text},
-		}, k.nodes))
-
-	k.refs = beam.ParDo(s, toRefs, beam.CoGroupByKey(s, files, anchors))
+	k.refs = beam.ParDo(s, toRefs, beam.CoGroupByKey(s, k.getFiles(), anchors))
 	return k.refs
 }
 
+func (k *KytheBeam) getFiles() beam.PCollection {
+	if !k.files.IsValid() {
+		fileNodes := beam.ParDo(k.s,
+			&nodes.Filter{
+				FilterByKind: []string{kinds.File},
+				IncludeFacts: []string{facts.Text, facts.TextEncoding},
+			}, k.nodes)
+		k.files = beam.ParDo(k.s, toFiles, fileNodes)
+	}
+	return k.files
+}
+
 func keyByPath(n *ppb.Node) (*spb.VName, *ppb.Node) {
 	return &spb.VName{Corpus: n.Source.Corpus, Root: n.Source.Root, Path: n.Source.Path}, n
 }
 
-func toRefs(p *spb.VName, file func(**ppb.Node) bool, anchor func(**ppb.Node) bool, emit func(*ppb.Reference)) error {
-	var fileNode *ppb.Node
-	if !file(&fileNode) {
+func toRefs(p *spb.VName, file func(**srvpb.File) bool, anchor func(**ppb.Node) bool, emit func(*ppb.Reference)) error {
```

Comment:
Please explain this change (and possibly update the PR description).  The PR description doesn't make it sound as though there's any bugs being fixed, and this seems pretty clearly a semantic change (i.e., not just new capabilities being added).

Reply:
This isn't fixing anything.  I changed the input from taking a raw `*ppb.Node` representing a file to take a `*srvpb.File`.  I did this to share the `*ppb.Node -> *srvpb.File` transform for the file decorations.